src/amplitude-client.js

```diff
@@ -1384,7 +1384,14 @@ var _generateApiPropertiesTrackingConfig = function _generateApiPropertiesTracki
  */
 AmplitudeClient.prototype._limitEventsQueued = function _limitEventsQueued(queue) {
   if (queue.length > this.options.savedMaxCount) {
-    queue.splice(0, queue.length - this.options.savedMaxCount);
+    const deletedEvents = queue.splice(0, queue.length - this.options.savedMaxCount);
```

Comment:
is `this.options.savedMaxCount` guaranteed to be set?

Reply:
Yup, the default config for it is [1000](https://github.com/amplitude/Amplitude-JavaScript/blob/f52288a82a9d07350712a84d83180f77295417ff/src/options.js#L77)